### PR TITLE
[Fix] Fix projects HieraSeg bug (fix Pull requests #2444)

### PR DESCRIPTION
**Pull request by AI-Tianlong**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Supplementary PR #2444 
Fix tiny bug and add loss_by_feat() to compute loss to train.
The inference process have verified to be accurate.
## Modification
- modify `sep_aspp_contrast_head.py` , add `loss_by_feat()` function to train(training still has bug, will fix in future😫) 
- fix  testing commands path error `bash tools/dist_test.sh projects/HieraSeg_project/` to `bash tools/dist_test.sh projects/HieraSeg/` at README.md 

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: projects/HieraSeg/decode_head/sep_aspp_contrast_head.py [hunk 1]
```diff
@@ -146,10 +154,45 @@ def predict_by_feat(self, seg_logits: Tensor,
...
+                data samples. It usually includes information such
+                as `img_metas` or `gt_semantic_seg`.
+            train_cfg (dict): The training config.
+
+        Returns:
+            dict[str, Tensor]: a dictionary of loss components
+        """
+        seg_logits = self.forward(inputs)
+        losses = self.loss_by_feat(seg_logits, batch_data_samples)
+        return losses
```
Comment (xiexinch) [inline projects/HieraSeg/decode_head/sep_aspp_contrast_head.py:173]: I suggest removing these codes since it's the same as the base class.

File: projects/HieraSeg/decode_head/sep_aspp_contrast_head.py [hunk 2]
```diff
@@ -100,27 +101,32 @@ def predict_by_feat(self, seg_logits: Tensor,
...
             seg_logit[:, 0:1] = seg_logit[:, 0:1] + \
                 seg_logit[:, 7] + seg_logit[:, 10]
             seg_logit[:, 1:5] = seg_logit[:, 1:5] + \
                 seg_logit[:, 8] + seg_logit[:, 11]
             seg_logit[:, 5:7] = seg_logit[:, 5:7] + \
                 seg_logit[:, 9] + seg_logit[:, 11]
-        elif seg_logit.size(1) == 25:
+
+        elif seg_logit.size(1) == 25:  # For Pascal_person dataset, 7 + 5
+            hiera_num_classes = 5
```
Comment (xiexinch) [inline projects/HieraSeg/decode_head/sep_aspp_contrast_head.py:129]: They are the same datasets?

File: projects/HieraSeg/decode_head/sep_aspp_contrast_head.py [hunk 3]
```diff
@@ -61,7 +62,7 @@ def __init__(self, proj: str = 'convmlp', **kwargs):
             dim_in=2048, norm_cfg=self.norm_cfg, proj=proj)
         self.register_buffer('step', torch.zeros(1))
 
-    def forward(self, inputs):
+    def forward(self, inputs) -> Tuple[Tensor]:
         """Forward function."""
         self.step += 1
         embedding = self.proj_head(inputs[-1])
```
Comment (xiexinch) [inline projects/HieraSeg/decode_head/sep_aspp_contrast_head.py:68]: The following part might replace with `super().forward(inputs)`

File: projects/HieraSeg/decode_head/sep_aspp_contrast_head.py (hunk 1 above)
Comment (AI-Tianlong) [reply projects/HieraSeg/decode_head/sep_aspp_contrast_head.py:173]: I agree with you😀

File: projects/HieraSeg/decode_head/sep_aspp_contrast_head.py (hunk 2 above)
Comment (AI-Tianlong) [reply projects/HieraSeg/decode_head/sep_aspp_contrast_head.py:129]: NO！I forget changing the annotation. I will fix now.😀
